packages/web/src/common/store/pages/signon/sagas.js

```diff
@@ -714,17 +716,33 @@ function* followCollections(collectionIds, favoriteSource) {
   }
 }
 
-function* followArtists() {
+/* This saga makes sure that artists chosen in sign up get followed accordingly */
+export function* completeFollowArtists(action) {
+  const accountId = yield select(accountSelectors.getUserId)
+  if (accountId) {
+    // If account creation has finished we need to make sure followArtists gets called
+    // Also we specifically request to not follow the defaults (Audius user, Hot & New Playlist) since that should have already occurred
+    yield put(signOnActions.followArtists(action.userIds, true))
+  }
+  // Otherwise, Account creation still in progress and followArtists will get called already, no need to call here
+}
+
+function* followArtists(action) {
+  const { skipDefaultFollows } = action
   const audiusBackendInstance = yield getContext('audiusBackendInstance')
   const { ENVIRONMENT } = yield getContext('env')
-  const defaultFollowUserIds = yield call(getDefautFollowUserIds)
+  const defaultFollowUserIds = skipDefaultFollows
+    ? []
+    : yield call(getDefautFollowUserIds)
   yield call(waitForWrite)
   try {
     // Auto-follow Hot & New Playlist
-    if (ENVIRONMENT === 'production') {
-      yield fork(followCollections, [4281], FavoriteSource.SIGN_UP)
-    } else if (ENVIRONMENT === 'staging') {
-      yield fork(followCollections, [555], FavoriteSource.SIGN_UP)
+    if (!skipDefaultFollows) {
```

Comment:
when would you want to skip default follows? oh i see this is for the second time around?

Reply:
Yeah since I'm calling the same "sign up focused" follow artists saga here I was running into a quirk where it triggered the default follows twice (resulted in an incorrect following count and extra transactions to re-follow the default stuff).
So yeah I just added a lil flag for my new saga to use